### PR TITLE
CameraXBasic updated to use 1.0.0-alpha10

### PR DESCRIPTION
**Pull request by syslogic**
Have updated the dependencies to latest version and fixed the `ImageCapturedCallback`. There's still `preview?.targetRotation` - which is immutable `val targetRotation`. Not sure if one would have to build a new `preview` on configuration change, since this value is only accessible through the `Builder`.

### CODE REVIEW

File: CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/CameraXBasic.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package com.android.example.cameraxbasic
+
+import android.app.Application
+import androidx.camera.camera2.Camera2Config
+import androidx.camera.core.CameraXConfig
+
+/** It implements CameraXConfig.Provider */
+class CameraXBasic: Application(), CameraXConfig.Provider {
```
Comment (SalvatoreT) [inline CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/CameraXBasic.kt]: You don't need to make an Application class that extends `CameraXConfig.Provider` for `alpha10`. That was a change between alpha 9 and 10.

**Comment by syslogic** [comment]
The current situation is that it takes a picture, saves it to `MediaStore` and does not crash while doing so. `outputDirectory` and `outputDirectory.listFiles` is currently not in use anymore. Or could one get `Uri` from `androidx.core.content.FileProvider`?

**Comment by owahltinez** [comment]
Hey @syslogic thanks for this PR, I'm glad to see we didn't scare you away after the last one 🙂

I think that you need to rebase with respect to the latest commit from our master branch, it looks like not everything is up to date.

**Comment by syslogic** [comment]
@owahltinez Can rebase and add a new PR - while the mere question is: whether to use a `ContentProvider` or not. I've only added constant `USE_CONTENT_PROVIDER` to demonstrate the two available options, where one of them fundamentally changes how the app is storing the images.

**Comment by owahltinez** [comment]
Good question about the content provider. I would say that it definitely belongs to a different PR, so we can discuss it further there. I'm not against implementing the content provider altogether, but we evaluated the idea when the sample was first written and opted to use the MediaScanner instead -- since it registers the media files all the same and it is much simpler to use (see #144 for some additional context). I would definitely prefer to either explicitly use content providers or not at all, adding the constant makes the code pretty difficult to understand for someone new to the idea.

**Comment by syslogic** [comment]
I'll try to keep the way images are being saved as before.

An alternate/different implementation of the `CameraFragment` might be better, so that both ways are being demonstrated apart from each other; think it would only need to override 2-3 methods - but the `ContentProvider` likely would show all images takes, not only those from the application.

`google/mediapipe`  also doesn't work with the current API: [472](https://github.com/google/mediapipe/issues/472)

**Pull request closed without merging**